src/viewport/db_metrics.py

```
@@ -0,0 +1,65 @@
+    pool_name: str,
+    long_checkout_seconds: float = LONG_CONNECTION_CHECKOUT_SECONDS,
+) -> None:
+    """Record actual connection checkout durations for one SQLAlchemy pool."""
+    sync_engine = engine.sync_engine if isinstance(engine, AsyncEngine) else engine
+    pool = sync_engine.pool
+    checkout_started_key = f"viewport_checkout_started:{pool_name}"
+    checked_out = DB_CONNECTIONS_CHECKED_OUT.labels(pool=pool_name)
+    checkout_duration = DB_CONNECTION_CHECKOUT_DURATION_SECONDS.labels(pool=pool_name)
+
+    def on_checkout(
+        _dbapi_connection: Any,
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_dbapi_connection`

(ANN401)